### ISSUE TITLE
(f (3)) != (f(3))

### ISSUE THREAD

**Issue by vird** [jashkenas/coffeescript#4113]
Function call with brackets make int const float const. So f (3) translated to f(3.)

```
 f = (t)->t
 alert (f (3)) != (f(3))
```

Can't imagine example when execution goes different ways, but extra dot is strange.

**Comment by michaelficarra** [comment]
JavaScript only has doubles, so they're the same. The extra dot is weird though, and probably related to us compiling `(0).a` to `0..a`.

**Comment by GeoffreyBooth** [comment]
Not sure what the issue is. The example above returns `false` for me, as I’d expect. I rewrote it to make it easier for me to understand, and [I still get expected results](http://coffeescript.org/v2/#try:f%20%3D%20(t)-%3E%20t%0A%0Aa%20%3D%20f%20(3)%0Ab%20%3D%20f(3)%0A%0Aconsole.log%20a%20is%20b%20%23%20true):

```coffee
f = (t)-> t

a = f (3)
b = f(3)

console.log a is b # true
```

**Comment by lydell** [comment]
@GeoffreyBooth The issue was that there was an unnecessary dot in the compiled JS: `f(3.);` (it didn't cause any harm but it was of course not meant to be there). Luckily, this seems to have been fixed in 1.11.0:

```
$ git checkout 1.10.0
$ ./bin/coffee -bpe 'f (3)'
f(3.);
$ git checkout 1.11.0
$ ./bin/coffee -bpe 'f (3)'
f(3);
```
